mesonbuild/backend/ninjabackend.py

```diff
@@ -746,7 +746,7 @@ def get_target_source_can_unity(self, target, source):
                 return False
         return True
 
-    def create_target_source_introspection(self, target: build.Target, comp: compilers.Compiler, parameters, sources, generated_sources):
+    def create_target_source_introspection(self, target: build.Target, comp: compilers.Compiler, parameters, sources, generated_sources, unity_sources=None):
```

Comment:
Might be nice to add a type annotation to unity_sources

Reply:
Maybe it should just default to an empty array? What do you think?

Reply:
Annotation added. You dont want to use mutable type as default value... `None` is fine.